File: python/armstrong-numbers/armstrong_numbers.py
def is_armstrong_number(number):
    str = f'{number}'
    length = len(str)
    digits = []
    for i in range(0, length):
        digits.append(int(str[i]))
    sum = 0
    for n in digits:
        sum += n ** length
    return sum == number

File: c/armstrong-numbers/armstrong_numbers.c
#include "armstrong_numbers.h"
#include "stdbool.h"
#include "math.h"
#include "stdio.h"

bool is_armstrong_number(int candidate) {
  int sum = 0;
  int num = candidate;
  int digits = 0;
  while (num > 0) {
    num /= 10;
    digits++;
  }
  num = candidate;
  while (num > 0) {
    sum += pow(num % 10, digits);
    num /= 10;
  }
  return sum == candidate;
}